### DISCUSSION TITLE
Failed to find a working CUDA architecture.

### DISCUSSION THREAD

**Discussion by Sven-vh** [NVlabs/instant-ngp#190] [Q&A]
When I was running: cmake . -B build

I received this error:
`-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
-- The CUDA compiler identification is NVIDIA 11.6.55
CMake Error at C:/Program Files/CMake/share/cmake-3.19/Modules/CMakeDetermineCUDACompiler.cmake:559 (message):
  Failed to find a working CUDA architecture.
Call Stack (most recent call first):
  CMakeLists.txt:11 (PROJECT)

And here are CMakeOutput.log and CMakeError.log:

Does anyone have an idea how to fix this?
The solution on this page https://github.com/NVlabs/instant-ngp#troubleshooting-compile-errors
also doesn't work
[CMakeError.log](https://github.com/NVlabs/instant-ngp/files/8103201/CMakeError.log)
[CMakeOutput.log](https://github.com/NVlabs/instant-ngp/files/8103202/CMakeOutput.log)

**Comment by Tom94** [comment]
If CMake can't detect CUDA, this means a compatibility mismatch between Visual Studio and CUDA's version.

In my experience, this can be resolved by upgrading visual studio through the visual studio installer
![image](https://user-images.githubusercontent.com/4923655/154833085-2d2a9e1f-5578-4a81-8036-037220120e15.png)

And subsequently re-installing CUDA & CMake.

Reboot afterwards and then fingers crossed that it'll detect correctly.

**Reply by Tom94** [reply]
Unfortunately I don't have any further ideas. Only that CUDA 11.6 definitely works with a 1660, so you don't need to downgrade. The problem must be something else.

It might be worth checking out the minimal compiling examples in `.github/workflows/main.yml` and trying to work out what's different about your local system. They run with every commit, confirming that the codebase works on a range of architectures / OSs and you can find the corresponding build logs here: https://github.com/NVlabs/instant-ngp/actions/runs/1864113932

**Reply by Sven-vh** [reply]
So I tried installing CUDA 11.6 but received this error. Searching it up online didn't help me.
![image](https://user-images.githubusercontent.com/54216207/154836767-ba136d12-315c-4a5e-b2e3-9811506b3684.png)
